src/components/Switch.js

```diff
@@ -1,44 +1,36 @@
-import React, { Component } from 'react'
-import styled from 'styled-components'
-import Button from "./Button";
-
+import { Component } from 'react'
 
 class Switch extends Component {
-    constructor(props) {
+    constructor (props) {
         super(props)
-        this.state = {
-            ...props.buttons.map(prop => prop.name)
-        }
+        this.state = {}
+    }
+
+    isActive = (prop) => {
+        return this.state[prop]
     }
+
     setCurrentState = (prop) => {
-        this.setState({ [prop]: !this.state[prop]})
         if (this.props.oneValue) {
-            for (let objProp in this.state) {
-                if (this.state[objProp]) {
-                    this.setState({ [objProp]: false})
-                }
-            }
+            this.setState({
```

Comment:
is this stage going to stay here in a stateful component, or are we planning to move it to a UI reducer?

Reply:
I wondered about it but @kgajowy said that we should not move logic to Redux, and it sounds like a good idea so I left logic inside the component :)

Reply:
what was his reason for not moving the logic to Redux?